core/trino-main/src/main/java/io/trino/execution/buffer/SpoolingExchangeOutputBuffer.java

```diff
@@ -22,6 +22,7 @@
 import io.trino.spi.exchange.ExchangeSink;
```

Comment:
Some test for `SpoolingExchangeOutputBuffer.java` state transitions would be really great

Reply:
I was thinking whether it is worth adding a test. It seems that such a test will rather be very shallow and won't test much.

Reply:
I think with the futures that test is now justified. It would document contract of valid state transitions

Reply:
Fair enough, especially now when it is guaranteed that the finish / abort method won't be called twice. Let me add a test